### DISCUSSION TITLE
Unable to access IMessage from Google Protobuffer

### DISCUSSION THREAD

**Discussion by Flechaa** [vfsfitvnm/frida-il2cpp-bridge#220] [Q&A]
![image](https://user-images.githubusercontent.com/50523969/189770948-89af8cdd-4f16-4314-ba3c-d72009bb6d91.png)

```ts
const CSharpAssembly = Il2Cpp.Domain.assembly("Assembly-CSharp").image;
const toTrace = CSharpAssembly.class("FLENDPBBDND");
toTrace.method("FJCOHKEGDFK").implementation = function() {
    const IMessageArray = this.field<Il2Cpp.Array>("PDGCDCIGBHE").value;
    for (let i = 0; i < IMessageArray.length; i++) {
        const element = IMessageArray.get(i) as Il2Cpp.Object;
        // this here causes abort was called
        console.log(element.method("ToString").invoke());
    }
    return this.method("FJCOHKEGDFK").invoke(...arguments);
}
```

When I attempt to read the IMessage I get this exception:

```js
il2cpp: System.NullReferenceException: Object reference not set to an instance of an object.
  at CHGOEJKCHON.OCOINFBPGMO () [0x00000] in <00000000000000000000000000000000>:0
  at Google.Protobuf.JsonFormatter.Format (Google.Protobuf.IMessage message) [0x00000] in <00000000000000000000000000000000>:0

Error: abort was called
    at NativeFunction.<anonymous> (<anonymous>)
    at C.invokeRaw (/frida/repl-2.js:375:14208)
    at Oe.e.method.implementation (/frida/repl-2.js:375:37627)
    at InvocationContext.i (/frida/repl-2.js:375:12750)
  ```

I also tried only acessing the method name but it gives me an access violation, I am not sure why this happens, all the IMessage instances that I try to access seem to have the same issue.
What could be the issue?

**Comment by vfsfitvnm** [comment]
Many details are missing, however I guess the game just throws an exception (this is not your fault). What happens if you keep:
```ts
Il2Cpp.installExceptionListener("all");
```
and remove the interception of that method completely?

**Reply by Flechaa** [reply]
It only throws this exception when I add

```
console.log(element.method("ToString").invoke())
```

removing this line or the interception of the method no longer throws any exception, which details am I missing?

**Reply by vfsfitvnm** [reply]
You aren't missing anything. I think `element` or one of its field is `null`, so the stringifier throws an exception

**Reply by Flechaa** [reply]
When I log the array it does say it's null but when I log element.method("ToString") it tells me which class implements IMessage
